pallets/subtensor/src/subnets/weights.rs

```diff
@@ -35,16 +35,27 @@ impl<T: Config> Pallet<T> {
             Error::<T>::CommitRevealDisabled
         );
 
+        // Check that the validator is not committing too frequently (weights_rate_limit check)
         ensure!(
-            Self::can_commit(netuid, &who),
+            Self::check_rate_limit_for_commit(netuid, &who),
```

Comment:
If you are using rate_limits , please follow the new pattern in utils/rate_limiting

Reply:
I guess it looks difference concept each other

Reply:
Please look at these 

- rate limit declaration : https://github.com/DedicatedDev/subtensor/blob/datura/feat/update-weight-commit-logic/pallets/subtensor/src/utils/rate_limiting.rs#L28
- how is it applied : https://github.com/DedicatedDev/subtensor/blob/datura/feat/update-weight-commit-logic/pallets/subtensor/src/staking/set_children.rs#L249-L277

Reply:
getting an error like that is the most confusing thing ever. Please return a very detailed error, saying "too early" or "too late" and detailing the exact extrinsic ids and block numbers involved so that people who need to figure out why a validator failed to set weights have something to go on

Reply:
@distributedstatemachine  I don't think so. 
Please check this. https://github.com/opentensor/subtensor/blob/58c1f51cc9f0716545fbf99c10c8d761d19dd472/pallets/subtensor/src/subnets/weights.rs#L321
It's similar things. not global feature

Reply:
@ppolewicz  "too early" | "too late' it has meaning? 
This is just checking weight set_rate_limit, 
If they did not make a quick update, that's possible to make a commit or not, impossible. 
So current error message enough for this action

Reply:
it's just the implementation of error parts, so treating it in other PR might be better. 
In macros error implementation, there are only simple default error implementations, I don't want to change this implementation in this PR so we can consider this in a separate PR. maybe you can open your PR for this. 
Thanks

Reply:
Sorry i missed this , @DedicatedDev . Can you read the whole rate limiting file , we have per subnet // global limits to choose from . In `set_children` , we use a subnet limit , but the global on is there too.